src/Runner.Worker/ActionManager.cs

```diff
@@ -101,38 +101,41 @@ public async Task<PrepareResult> PrepareActionsAsync(IExecutionContext execution
             IEnumerable<Pipelines.ActionStep> actions = steps.OfType<Pipelines.ActionStep>();
             executionContext.Output("Prepare all required actions");
             var result = await PrepareActionsRecursiveAsync(executionContext, state, actions, depth, rootStepId);
-            if (state.ImagesToPull.Count > 0)
+            if (!FeatureFlagManager.IsContainerHooksEnabled(executionContext.Global.Variables))
```

Comment:
I think we still want to do this after the feature flag is enabled when you are not using container hooks.

Reply:
That should also be checked in `FeatureManager. IsContainerHooksEnabled`